src/nvim/lua/treesitter.c

```diff
@@ -175,7 +175,8 @@ int tslua_add_language(lua_State *L)
   }
 
   uint32_t lang_version = ts_language_version(lang);
-  if (lang_version < TREE_SITTER_MIN_COMPATIBLE_LANGUAGE_VERSION) {
+  if (lang_version < TREE_SITTER_MIN_COMPATIBLE_LANGUAGE_VERSION
+      || lang_version > TREE_SITTER_LANGUAGE_VERSION) {
```

Comment:
In `ts_parser_set_language` the very same check is performed, so we try to detected this error as early as possible.